### PR TITLE
Free Parking Feature

### PR DESCRIPTION
**Pull request by YurouTang**
Basic Implementation of free parking implementation
Test: filter f/ [DAY] [START TIME] [END TIME] 
e.g. filter f/ sun 10.00AM 9.00PM

### CODE REVIEW

File: src/main/java/seedu/address/model/carpark/CarparkHasFreeParkingPredicate.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package seedu.address.model.carpark;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.function.Predicate;
+
+/**
+ * To be added
```
Comment (ZhuanYu) [inline src/main/java/seedu/address/model/carpark/CarparkHasFreeParkingPredicate.java]: Add javadoc

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 2]
```diff
@@ -23,22 +27,37 @@
             + "Example: " + COMMAND_WORD + "f/TRUE";
 
     //public static final String MESSAGE_FILTER_CARPARK_SUCCESS = "Filtered Carparks.";
```
Comment (ZhuanYu) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java]: Space carparks

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 3]
```diff
@@ -5,12 +5,16 @@
...
+import java.util.function.Predicate;
+
 import seedu.address.commons.core.Messages;
 import seedu.address.logic.CommandHistory;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
+import seedu.address.model.carpark.CarparkHasFreeParkingPredicate;
 import seedu.address.model.carpark.CarparkHasNightParkingPredicate;
 
+
```
Comment (ZhuanYu) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java]: Javadoc

File: src/main/java/seedu/address/model/carpark/CarparkHasFreeParkingPredicate.java (hunk 1 above)
Comment (YurouTang) [reply src/main/java/seedu/address/model/carpark/CarparkHasFreeParkingPredicate.java]: Change made

File: src/main/java/seedu/address/logic/commands/FilterCommand.java (hunk 2 above)
Comment (YurouTang) [reply src/main/java/seedu/address/logic/commands/FilterCommand.java]: Change made

File: src/main/java/seedu/address/logic/commands/FilterCommand.java (hunk 3 above)
Comment (YurouTang) [reply src/main/java/seedu/address/logic/commands/FilterCommand.java]: Change made

**Pull request merged**